tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/configurables/ConfigurableTest.java

```diff
@@ -144,159 +127,126 @@ public void testInvalidTomlFile() throws BallerinaTestException {
-        bMainInstance.runMain("run", new String[]{"main"}, null, new String[]{},
-                              new LogLeecher[]{errorLeecher}, projectPath.toString());
-        errorLeecher.waitForText(5000);
+    @DataProvider(name = "negative-tests")
+    public Object[][] getNegativeTests() {
+        return new Object[][]{
+                {"no_module_config", "Value not provided for required configurable variable 'main:stringVar'"},
+                {"invalid_org_name", "Value not provided for required configurable variable 'main:stringVar'" },
+                {"invalid_org_structure", "invalid module structure found for module 'testOrg.main'. " +
+                        "Please provide the module name as '[testOrg.main]'" },
+                {"invalid_module_structure", "invalid module structure found for module 'main'. " +
+                        "Please provide the module name as '[main]'" },
+                {"invalid_sub_module_structure", "invalid module structure found for module 'foo'. " +
+                        "Please provide the module name as '[foo]'" },
+                {"required_negative", "Value not provided for required configurable variable 'main:stringVar'"},
+                {"primitive_type_error", "invalid type found for variable 'main:intVar', expected TOML type " +
+                        "for type 'int' is 'INTEGER', found 'DOUBLE'"},
+                {"primitive_structure_error", "invalid TOML structure found for variable 'main:intVar', " +
+                        "expected structure is 'KEY_VALUE', found 'TABLE'"},
+                {"array_type_error", "invalid type found for variable 'main:intArr', expected TOML type " +
+                        "for type 'int[] & readonly' is 'ARRAY', found 'STRING'"},
+                {"array_structure_error", "invalid TOML structure found for variable 'main:intArr', " +
+                        "expected structure is 'KEY_VALUE', found 'TABLE'"},
+                {"array_element_structure", "invalid TOML structure found for variable 'main:intArr[2]', " +
+                        "expected structure is 'INTEGER', found 'ARRAY'"},
+                {"array_multi_type", "invalid TOML structure found for variable 'main:intArr[1]', " +
+                        "expected structure is 'INTEGER', found 'STRING'"},
+                {"additional_field", "Additional field 'scopes' provided for configurable variable 'main:testUser'" +
+                        " of record 'main:AuthInfo' is not supported"},
+                {"missing_record_field", "Value not provided for non-defaultable required field 'username' of record" +
+                        " 'main:AuthInfo' in configurable variable 'main:testUser'"},
+                {"record_type_error", "invalid type found for variable 'main:testUser', expected TOML type for type " +
+                        "'main:(testOrg/main:0.1.0:AuthInfo & readonly)' is 'TABLE', found 'KEY_VALUE'"},
```

Comment:
'user' is expected to be a recordType, but found 'string'. Value for 'user' is required to be defined as toml table in the Config.toml